tests/UniswapOracleTest.js

```diff
@@ -0,0 +1,292 @@
+    "1592411459",
+    "8940376039738985960413226930501944964252039",
+    "20280827635915824975600233241328957701",
+  ]);
+
+  // Initialize WBTC_ETH pair with values from mainnet
+  const wbtc_eth_pair = await deploy("MockUniswapTokenPair", [
+    "3647713616",
+    "1482718962516231732009",
+    "1592423618",
+    "5649104323398751783577661386954839177549366239904392",
+    "32429167300960843298079679680",
+  ]);
+
+  const uniswapOracle = await deploy("UniswapOracle", [0,
+    {
+      USDC_ETH_pair: usdc_eth_pair._address,
+      DAI_ETH_pair: dai_eth_pair._address,
+      BAT_ETH_pair: bat_eth_pair._address,
+      REP_ETH_pair: rep_eth_pair._address,
+      WETH_ZRX_pair: weth_zrx_pair._address,
+      WBTC_ETH_pair: wbtc_eth_pair._address,
+    },
+  ]);
+
+  async function getPrice(symbol) {
+    return call(uniswapOracle, "getPrice", [symbol]);
+  }
+
+  async function update(symbol) {
+    return send(uniswapOracle, "update", [symbol]);
+  }
+
+  function sleep(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  return {
+    usdc_eth_pair,
+    dai_eth_pair,
+    uniswapOracle,
+    update,
+    getPrice,
+    sleep,
+  };
+}
+
+describe("UniswapOracle", () => {
+  let update;
+  let getPrice;
+  let sleep;
+
+  describe("get TWAP asset prices", () => {
+    beforeEach(async (done) => {
+      ({
+        usdc_eth_pair,
+        dai_eth_pair,
+        uniswapOracle,
+        update,
+        getPrice,
+        sleep,
+      } = await setup());
+      done();
+    });
+
+    it("USDC_ETH pair price", async () => {
+      const price1 = await getPrice("ETH");
+      expect(price1.price0Average).numEquals(0);
+      expect(price1.price1Average).numEquals(0);
+      
+      await sleep(2000);
+      await update("ETH");
+      await sleep(2000);
+      await update("ETH");
+
+      const price2 = await getPrice("ETH");
+      expect(price2.price0Average).toBe("4336069182197935199725599365");
+      expect(price2.price1Average).toBe("230623626");
+
+      // Real world prices
+      // 1 USDC = 0.0043360691821979352 ETH
+      expect(
+        new BigNumber(price2.price0Average)
+          .div(new BigNumber(1e18))
+          .div(new BigNumber(1e12))
+          .toString()
+      ).toBe("0.0043360691821979352");
+      // 1 ETH = 230.623626
+      expect(
+        new BigNumber(price2.price1Average)
+          .div(new BigNumber(1e18))
+          .multipliedBy(new BigNumber(1e12))
+          .toString()
+      ).toBe("230.623626");
+    });
+
+    it("DAI_ETH pair price", async () => {
+      const price1 = await getPrice("DAI");
+      expect(price1.price0Average).numEquals(0);
+      expect(price1.price1Average).numEquals(0);
+
+      await sleep(2000);
+      await update("DAI");
+      await sleep(2000);
+      await update("DAI");
+
+      const price2 = await getPrice("DAI");
+      expect(price2.price0Average).toBe("4405985580676962");
+      expect(price2.price1Average).toBe("226963974731472890029");
```

Comment:
as mentioned, we only care about one of these prices, right?

Reply:
Yes, but I am adding check for 2, why not :)